provisioner/helper_kernel_nfs_server.go

```diff
@@ -212,6 +216,9 @@ func (p *Provisioner) createDeployment(nfsServerOpts *KernelNFSServerOptions) er
 		WithPodTemplateSpecBuilder(
 			pts.NewBuilder().
 				WithLabelsNew(nfsDeployLabelSelector).
+				WithSecurityContext(&corev1.PodSecurityContext{
+					FSGroup: nfsServerOpts.fsGroup,
```

Comment:
when the storage class doesn't specify this value, how is it handled?

Reply:
When users don't specify a value then we are setting `nil` in FSGroup of the pod which will act as default behavior(i.e volume can only accessible by root user).

https://github.com/mittachaitu/dynamic-nfs-provisioner/blob/non-root-access/provisioner/config.go#L181-L193